### PR TITLE
T/1063: Introduce `Selection#isEntireContentSelected( element )`

### LINKED ISSUE #1063: Introduce `Selection#isEntireContentSelected( element )`
This method should checks whether the entire content for specified element is selected.

More details: https://github.com/ckeditor/ckeditor5-typing/pull/111#issuecomment-317653979

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Added `Selection#isEntireContentSelected( element )`. Closes #1063.

### CODE REVIEW

File: src/model/selection.js [hunk 1]
```diff
@@ -625,6 +625,20 @@ export default class Selection {
 	}
 
 	/**
+	 * Checks whether the entire content in specified element is selected.
```
Comment (Reinmar) [inline src/model/selection.js]: This description is not accurate. In such case, the `<b>`'s content is also fully selected:

```
<div>[x<b>y</b>z]</div>
```

It's hard to explain this in just one paragraph, but we should try with something like:

Checks whether the selection contains the entire content of the given element. This means that selection must start at a position {@link module:engine/model/position~Position#isTouching touching} the element's start and ends at position touching the element's end.

File: src/model/selection.js [hunk 2]
```diff
@@ -625,6 +625,20 @@ export default class Selection {
...
+	 *
+	 * @param {module:engine/model/element~Element} element
+	 * @returns {Boolean}
+	 */
+	isEntireContentSelected( element ) {
```
Comment (Reinmar) [inline src/model/selection.js]: This element should default to the root of the selection's root (`this.anchor.root` will be fine).

File: tests/model/selection.js [hunk 3]
```diff
@@ -1237,4 +1237,56 @@ describe( 'Selection', () => {
 			} );
 		} );
 	} );
+
+	describe( 'isEntireContentSelected()', () => {
```
Comment (Reinmar) [inline tests/model/selection.js]: Missing tests:

* empty content,
* empty selection at a boundary of non-empty content

Comment (pomek) [inline tests/model/selection.js]: Added.

**Comment by pomek** [comment]
I guess it's ready for review once again.

**Pull request merged**